lib/std/mem.zig

```diff
@@ -853,8 +853,7 @@ pub fn indexOf(comptime T: type, haystack: []const T, needle: []const T) ?usize
 
 /// Find the index in a slice of a sub-slice, searching from the end backwards.
 /// To start looking at a different index, slice the haystack first.
-/// TODO is there even a better algorithm for this?
-pub fn lastIndexOf(comptime T: type, haystack: []const T, needle: []const T) ?usize {
+fn lastIndexOfNaive(comptime T: type, haystack: []const T, needle: []const T) ?usize {
```

Comment:
`lastIndexOfLinear` is a better name IMO.

Reply:
fixed

Reply:
Naive string search is not linear. Its worst case is O(n*m) where n is `needle.len` and m is `haystack.len`.

Reply:
It's a linear scan of the input with no fancy skipping.

Reply:
It runs `mem.eql` in a loop. I think putting 'linear' in the name is misleading. I prefer 'naive'.

Reply:
Names shouldn't reflect the big O complexity, this is a linear scan over a sequence hence my suggestion. I'm not [alone](https://android.googlesource.com/platform/external/v8/+/77008027c3252aa4e9d5f03f66c42e435bd69aee/src/string-search.h#253) in using this naming convention.